pkg/storage/postgres/insert.go

```diff
@@ -144,14 +144,16 @@ func insertVCSIfNotExist(ctx context.Context, db dbConn, revision string, isDirt
 
 func insertInputFilesIfNotExist(ctx context.Context, db dbConn, inputs []*storage.InputFile) ([]int, error) {
 	const stmt1 = `
-           INSERT INTO input_file (path, digest)
-	   VALUES
-`
+	     WITH data (path, digest) AS (
```

Comment:
Isn't this exactly the same as sort sorting `inputs` by `inputs.Path` and `inputs.Digest` on go, then execute the sql statement as before? I think the code would read a lot better, and the effect on avoiding the deadlock would be the same, no? I mean, this code looks totally correct, but it took me a few minutes to understand it.

Reply:
Yes, it is exactly the same and I agree, it's easier to understand when sorting the slice in Go.
Performance-wise it should also be fine. The additional required memory of the slice copy also does not matter much, it only contains pointers.
I changed it.

thanks for the suggestion!